### ISSUE TITLE
SIGSEGV on "Assemble"

### ISSUE THREAD

**Issue by itsacoderepo** [Vector35/binaryninja-api#981]
Binary Ninja Version: 1.1.1094-dev, b8f790be
Platform: Arch Linux

Reproduction steps:
1. Start binary ninja
2. Press Ctrl+N
3. Change from "Hex Editor" to "Linear dissassembly"
4. Press Ctrl+Alt+A ("View" -> "Assemble")

```
Thread 1 "binaryninja" received signal SIGSEGV, Segmentation fault.
0x0000000000547ca7 in BinaryNinja::BasicBlock::GetArchitecture() const ()
(gdb) bt
#0  0x0000000000547ca7 in BinaryNinja::BasicBlock::GetArchitecture() const ()
#1  0x00000000004eaea0 in LinearView::getCurrentArchitecture() ()
#2  0x0000000000459eb4 in ViewFrame::assemble() ()
#3  0x0000000000528545 in MainWindow::qt_static_metacall(QObject*, QMetaObject::Call, int, void**) ()
#4  0x00007ffff44bbbc9 in QMetaObject::activate(QObject*, int, int, void**) () from binaryninja/libQt5Core.so.5
#5  0x00007ffff562f492 in QAction::triggered(bool) () from binaryninja/libQt5Widgets.so.5
#6  0x00007ffff5631990 in QAction::activate(QAction::ActionEvent) () from binaryninja/libQt5Widgets.so.5
#7  0x00007ffff56322ff in QAction::event(QEvent*) () from binaryninja/libQt5Widgets.so.5
#8  0x00007ffff563554c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from binaryninja/libQt5Widgets.so.5
#9  0x00007ffff563c897 in QApplication::notify(QObject*, QEvent*) () from binaryninja/libQt5Widgets.so.5
#10 0x00007ffff44909f8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from binaryninja/libQt5Core.so.5
#11 0x00007ffff4e53ae8 in QShortcutMap::dispatchEvent(QKeyEvent*) () from binaryninja/libQt5Gui.so.5
#12 0x00007ffff4e53ba9 in QShortcutMap::tryShortcut(QKeyEvent*) () from binaryninja/libQt5Gui.so.5
#13 0x00007ffff4e090e2 in QWindowSystemInterface::handleShortcutEvent(QWindow*, unsigned long, int, QFlags<Qt::KeyboardModifier>, unsigned int, unsigned int, unsigned int, QString const&, bool, unsigned short) () from binaryninja/libQt5Gui.so.5
#14 0x00007ffff4e2426c in QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) () from binaryninja/libQt5Gui.so.5
#15 0x00007ffff4e292f5 in QGuiApplicationPrivate::processWindowSystemEvent(QWindowSystemInterfacePrivate::WindowSystemEvent*) () from binaryninja/libQt5Gui.so.5
#16 0x00007ffff4e0515b in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from binaryninja/libQt5Gui.so.5
#17 0x00007fffee4779ab in ?? () from binaryninja/qt/platforms/../../libQt5XcbQpa.so.5
#18 0x00007ffff448f26a in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from binaryninja/libQt5Core.so.5
#19 0x00007ffff4497da4 in QCoreApplication::exec() () from binaryninja/libQt5Core.so.5
#20 0x000000000043c404 in main ()
```

**Comment by psifertex** [comment]
Repros on other platforms too, thanks for filing. 

Likely due to some recent UI changes to make a few dialogs no longer be modal.

**Comment by bpotchik** [comment]
Thanks for reporting. Fixed in build 1.1.1095-dev.
